astro_2.0/features/DaoDashboardHeader/components/hooks.ts

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState } from 'react';
```

Comment:
I would suggest to create hooks folder and place each hook in a file with proper name. `useJoinDao` for this particular case.